lib/generator/js.rb

```diff
@@ -271,6 +273,18 @@ def js_for_data_criteria(force_sources=nil)
         HQMF2JS::Generator.render_template('data_criteria', {'all_criteria' => @doc.specific_occurrence_source_data_criteria(force_sources).concat(@doc.all_data_criteria), 'measure_period' => @doc.measure_period})
       end
       
+      def js_for_variables()
+        variables_js = ""
+        variables_js += "hqmfjs.VARIABLES = function(patient, initialSpecificContext) {\n" 
+        @doc.source_data_criteria.each do |criteria|
+          if criteria.type == :derived
+            variables_js += "hqmfjs." + criteria.id + "(patient, initialSpecificContext);\n"
+          end
+        end
+        variables_js += "return false;\n}"
```

Comment:
Sorry, should have been more clear, I was talking about the generated code; does that need to return false?